### PR TITLE
Add Sphinx docs for the non-deprecated logging functions and classes

### PR DESCRIPTION
**Pull request by eric-wieser**
https://external-builds.readthedocs.io/html/cocotb/1412/library_reference.html#logging

### CODE REVIEW

File: cocotb/log.py [hunk 1]
```diff
@@ -125,6 +125,12 @@ def SimLog(name, ident=None):
 class SimLogFormatter(logging.Formatter):
     """Log formatter to provide consistent log message handling."""
 
+    # Removes the arguments from the base class. Docstring needed to make
+    # sphinx happy.
+    def __init__(self):
+        """ Takes no arguments. """
+        super().__init__()
```
Comment (eric-wieser) [inline cocotb/log.py]: @imphil, if you make the change I suggested in #1410, we could add the `style` argument here.

Comment (eric-wieser) [reply cocotb/log.py]: Nevermind, ignore that remark, I'm incorrect.

**Comment by eric-wieser** [comment]
@cmarqu: In particular, would appreciate suggestions about which heading this should fall under. Ok with you moving it later as part of a wider restructure too.

**Comment by cmarqu** [comment]
@eric-wieser The heading is fine IMO. What would be beneficial in addition is a cross-reference from https://external-builds.readthedocs.io/html/cocotb/1412/testbench_tools.html#logging

**Comment by eric-wieser** [comment]
@cmarqu: Updated

**Comment by eric-wieser** [comment]
And doc CI build is now done.

**Pull request merged**